### PR TITLE
New gles3 version: fix compilation + various bugs

### PR DESCRIPTION
**Pull request by craff**

### CODE REVIEW

File: packages/gles3/gles3.20160505.alpha/opam [hunk 1]
```diff
@@ -0,0 +1,28 @@
+opam-version: "1.2"
+name: "gles3"
+version: "20160505.alpha"
+maintainer: "Christophe Raffalli <raffalli@univ-savoie.fr>"
+available:  [ ocaml-version >= "4.02.0" & os != "darwin" ]
+authors:
+  "Christophe Raffalli <raffalli@univ-savoie.fr>, Alexandre Miquel<amiquel@fing.edu.uy>"
```
Comment (dsheets) [inline packages/gles3/gles3.20160505.alpha/opam]: `authors` field is a list so you can use `[ "name" "name2" ]`. With this syntax, both authors will be listed together always and it is difficult to process the metadata.

File: packages/gles3/gles3.20160505.alpha/opam [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+homepage: "http://lama.univ-savoie.fr/~raffalli/gles3"
+bug-reports: "raffalli@univ-savoie.fr"
```
Comment (dsheets) [inline packages/gles3/gles3.20160505.alpha/opam]: Could you please use `mailto:` here?

**Comment by dsheets** [comment]
Thanks for this update. There are just a couple metadata nits which are not blockers but would be nice to have fixed. This package won't build on CI so once you're happy with the metadata, we'll merge it.

Thanks!

**Comment by dsheets** [comment]
@craff would you like to update this? If not, I will just merge it as-is soon.

**Comment by craff** [comment]
No don't merge, I will do the update. 

By the way, why are you still using ubuntu 12.04 for most of the tests. There are now two more recent LTS out for ubuntu ?

File: packages/gles3/gles3.20160505.alpha/opam [hunk 3]
```diff
@@ -3,8 +3,10 @@ name: "gles3"
 version: "20160505.alpha"
 maintainer: "Christophe Raffalli <raffalli@univ-savoie.fr>"
 available:  [ ocaml-version >= "4.02.0" & os != "darwin" ]
+bug-reports:  "mailto:raffalli@univ-savoie.fr"
 authors:
-  "Christophe Raffalli <raffalli@univ-savoie.fr>, Alexandre Miquel<amiquel@fing.edu.uy>"
+  [ "Christophe Raffalli <raffalli@univ-savoie.fr>"
+    "Alexandre Miquel<amiquel@fing.edu.uy>" ]
 homepage: "http://lama.univ-savoie.fr/~raffalli/gles3"
 bug-reports: "raffalli@univ-savoie.fr"
```
Comment (dsheets) [inline packages/gles3/gles3.20160505.alpha/opam]: This one is extra.

File: packages/gles3/gles3.20160505.alpha/opam [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+name: "gles3"
+version: "20160505.alpha"
+maintainer: "Christophe Raffalli <raffalli@univ-savoie.fr>"
+available:  [ ocaml-version >= "4.02.0" & os != "darwin" ]
+bug-reports:  "mailto:raffalli@univ-savoie.fr"
+authors:
+  [ "Christophe Raffalli <raffalli@univ-savoie.fr>"
+    "Alexandre Miquel<amiquel@fing.edu.uy>" ]
+homepage: "http://lama.univ-savoie.fr/~raffalli/gles3"
+bug-reports: "raffalli@univ-savoie.fr"
```
Comment (dsheets) [inline packages/gles3/gles3.20160505.alpha/opam]: This one is extra.

**Comment by dsheets** [comment]
Ubuntu 12.04 is fixed by Travis, I think, but perhaps we can update it. @yallop?

**Comment by craff** [comment]
I got the mail to wrong ... by grep :
camlp5/camlp5.6.12/opam:bug-reports: "mailto:daniel.de_rauglaudre@inria.fr"
camlp5/camlp5.6.13/opam:bug-reports: "mailto:daniel.de_rauglaudre@inria.fr"
camlp5/camlp5.6.14/opam:bug-reports: "mailto:daniel.de_rauglaudre@inria.fr"
gles3/gles3.20160505.alpha/opam:bug-reports:  "mailto:raffalli@univ-savoie.fr"
ocamlfind/ocamlfind.1.5.5/opam:bug-reports:  "mailto:gerd@gerd-stolpmann.de"
ocamlfind/ocamlfind.1.5.6/opam:bug-reports:  "mailto:gerd@gerd-stolpmann.de"
ocamlfind/ocamlfind.1.6.1/opam:bug-reports:  "mailto:gerd@gerd-stolpmann.de"
portia/portia.0.1/opam:bug-reports: "mailto:rixed-opam@happyleptic.org"
unison/unison.2.48.3/opam:bug-reports: "mailto:unison-users@yahoogroups.com "

So I do not understand where i am supposed to add a mailto: ? There are no other mailto in existing package ?

**Comment by yallop** [comment]
> Ubuntu 12.04 is fixed by Travis, I think, but perhaps we can update it. @yallop?

Eventually we'll switch to Docker-based builds.  In the meantime we could switch to [Trusty Beta](https://docs.travis-ci.com/user/ci-environment/), i.e. Ubuntu 14.04.

**Comment by dsheets** [comment]
@craff you have 2 `bug-reports` lines.

**Comment by yallop** [comment]
> In the meantime we could switch to Trusty Beta, i.e. Ubuntu 14.04.

See #6573.

**Comment by craff** [comment]
May be the "invalid field" message is a bit misleading for an duplicate key ...

**Comment by lefessan** [comment]
Would it be possible to move the depexts to a package `conf-gles3` or something like that ?

**Comment by dsheets** [comment]
Creating a `conf-` package can be done in another pull request. This one has been open almost 3 weeks for terribly minor details so merging.

Thanks for the contribution and sorry for the delay!

**Pull request merged**